### PR TITLE
Add ArgoCD dashboard link to GitOpsListPage

### PR DESCRIPTION
**Pull request by reginapizza**
This PR is to add a link to the Argo CD dashboard on the GitOpsListPage (see image below). 

Things to note: 
- The argocd URL is pulled from ApplicationMenu
- If ConsoleLink does not have an instance called `argocd` then the link will not appear
- The story for this can be found [here](https://issues.redhat.com/browse/GITOPS-386)
- This PR does *not* include changes to add the ArgoCD link to the GitOpsDetailsPage, that will be done in a later PR once ArgoCD application-specific URLs can be worked out. 

GitOpsListPage

### CODE REVIEW

**Review by divyanshiGupta — CHANGES_REQUESTED** [review]
@reginapizza Please add prettier extension to your code editor and format the code. Also add eslint extension if you haven't already to visualize lint errors in the editor itself. I see your tab size is also not correct please change it to 2.

File: frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx
Comment (reginapizza) [reply frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx:58]: if I change this code snippet to your suggestion, I now get an error for `href={aLink.spec.href}` that says that `aLink` is of type `never` so therefore I can't get property `spec`. I've never worked with this type before, did you mean to change it to that on purpose?

File: frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx
Comment (divyanshiGupta) [inline frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx:44]:
```suggestion
  const [consoleLinks] = useK8sWatchResource<K8sResourceKind[]>({
```

File: frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx
Comment (divyanshiGupta) [reply frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx:58]: This is happening because you havent added `K8sResourceKind[]` as a type here https://github.com/openshift/console/pull/7127/files#diff-2091e68385783885a22919dac9db27a22a1eed67754375a6633cd0754d63e414R44 and by default the type is 'K8sResourceCommon[]' and since here the link type is `K8sResourceKind` and hence there is a type mismatch and the function never reaches an endpoint/or execute and no value is returned and thus the type of `aLink` becomes `never`. You can read more about `never` here https://www.tutorialsteacher.com/typescript/typescript-never

**Review by divyanshiGupta — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx
Comment (divyanshiGupta) [inline frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx:73]: Nit: This component is self-closing since it is empty.

**Review by divyanshiGupta — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx
Comment (divyanshiGupta) [reply frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx:73]: These issues are captured by the linter but I think you still havent added eslint to your code editor. Also when you compile the code you can see the lint errors in your terminal as well. The frontend job is also failing on this PR because of these lint errors.

Comment (reginapizza) [reply frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx]: Thanks, I've updated this now but I do have both Prettier and ESLint installed and enabled so I'm not sure why it's not giving me errors for anything

Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx]: If you open your editor in root of console repository then it doesn't work correctly. You should open the editor in `console/frontend`. This could probably be a reason for the issue.

File: frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx:49]:
```suggestion
  const argocdLink = _.find(consoleLinks, (link: K8sResourceKind) => 
```

File: frontend/packages/dev-console/src/components/gitops/GitOpsListPage.scss
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/gitops/GitOpsListPage.scss:9]: You don't need to specifically add margin like this. You can use `hasGutter` prop on `Split` component.

File: frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx:69]:
```suggestion
              <Split className="odc-gitops-list-page-heading">
                {argocdLink && <SplitItem><ExternalLink href={aLink.spec.href} text="Argo CD" additionalClassName="odc-gitops-list-page-heading__argocd"/></SplitItem>}
                <SplitItem><DevPreviewBadge /></SplitItem>
              </Split>
```

File: frontend/packages/dev-console/src/components/gitops/GitOpsListPage.scss
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/gitops/GitOpsListPage.scss:5]: You probably don't need these styles if you use `<Split>` component along with `<SplitItem>`

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx
Comment (andrewballantyne) [reply frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx:13]: We have a backlog ticket to try and make this automatic / linter errored. Apologizes for this inconvenience @reginapizza

File: frontend/packages/dev-console/src/components/gitops/GitOpsListPage.scss
Comment (divyanshiGupta) [inline frontend/packages/dev-console/src/components/gitops/GitOpsListPage.scss:10]:
```suggestion
.odc-gitops-list-page-heading {
  align-items: center;
  font-size: 12px;
}

```

Comment (divyanshiGupta) [reply frontend/packages/dev-console/src/components/gitops/GitOpsListPage.scss:10]: Your tab size is still not correct. Please change it to 2.

File: frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx
Comment (divyanshiGupta) [inline frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx:73]:
```suggestion
            <SplitItem>
              {argocdLink && (
                <ExternalLink
                  href={argocdLink.spec.href}
                  text="Argo CD"
                />
              )}
            </SplitItem>
            <SplitItem>
              <DevPreviewBadge />
            </SplitItem>
```

**Review by divyanshiGupta — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx
Comment (divyanshiGupta) [reply frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx:73]: The `hasGutter` property was not adding the required space between the two items since you wrapped both the items in a single `SplitItem` comp. As suggested above divide the items into two `SplitItem` comps and remove the css for adding space between the two.

File: frontend/packages/dev-console/src/components/gitops/GitOpsListPage.scss
Comment (divyanshiGupta) [inline frontend/packages/dev-console/src/components/gitops/GitOpsListPage.scss:4]: Please add a new line at the end of the file

**Comment by divyanshiGupta** [comment]
@reginapizza code looks good to me. I will add lgtm after @lwrigh approves this PR.

**Comment by lwrigh** [comment]
@reginapizza Can you add a screenshot of what the Application Details/Application Environments page looks like now with the ArgoCD link in the stage columns?

**Comment by reginapizza** [comment]
@lwrigh the details page change will actually be in a separate PR, this is only for the list page just so we could get in the changes faster. The details page PR will be created today and I will add the pictures from it, hope this explanation helps. Let me know if you still want it here though!

**Comment by lwrigh** [comment]
> @lwrigh the details page change will actually be in a separate PR, this is only for the list page just so we could get in the changes faster. The details page PR will be created today and I will add the pictures from it, hope this explanation helps. Let me know if you still want it here though!

@reginapizza That sounds good to me! Feel free to tag me on the separate PR. This one looks good to me then!

**Review by rohitkrai03 — APPROVED** [review]
/approve

**Pull request merged**
